### PR TITLE
dotCMS/core#13723 Content types UI issues

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 1]
```diff
@@ -140,22 +122,15 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
...
-        if (this.form.get('searchable')) {
-            this.handleCheckValues(this.form.get('searchable'));
-        }
-        if (this.form.get('listed')) {
-            this.handleCheckValues(this.form.get('listed'));
-        }
-        if (this.form.get('unique')) {
-            this.handleCheckValues(this.form.get('unique'));
-        }
+        [this.form.get('searchable'), this.form.get('listed'), this.form.get('unique')]
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: This was a cyclomatic complexity fix.

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 2]
```diff
@@ -189,13 +162,24 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
         }
 
         this.handleDisabledRequired(propertyValue);
+        this.handleDisabledIndexed(true);
     }
 
     private handleDisabledIndexed(disable: boolean): void {
-        disable ? this.form.get('indexed').disable() : this.form.get('indexed').enable();
+        if (this.form.get('indexed')) {
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: I think this should add a new test

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 3]
```diff
@@ -189,13 +162,24 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
...
+            disable ? this.form.get('indexed').disable() : this.form.get('indexed').enable();
+        }
     }
 
     private handleDisabledRequired(disable: boolean): void {
-        disable ? this.form.get('required').disable() : this.form.get('required').enable();
+        if (this.form.get('required')) {
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: I think this should add a new test

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts (hunk 2 above)
Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: If we had tests...

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts (hunk 3 above)
Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: If we had tests... I think we should refactor the way we do that thing of the checkboxes, but we don't have time right now.

Comment (freddyucv) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: but I think we should create the new test anyway

Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: Yep, you're righ, I'm on it

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts [hunk 4]
```diff
@@ -88,70 +131,72 @@ describe('ContentTypeFieldsPropertiesFormComponent', () => {
...
+        beforeEach(() => {
+            spyOn(mockFieldPropertyService, 'getProperties').and.returnValue(['searchable', 'required', 'unique', 'indexed', 'listed']);
+            spyOn(mockFieldPropertyService, 'existsComponent').and.returnValue(true);
+            startHostComponent();
         });
 
-        xit('should auto select and disable require checkbox', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed true when select user searchable', () => {
+            comp.form.get('indexed').setValue(false);
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts]: We need a test when indexed is undefined

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts [hunk 5]
```diff
@@ -88,70 +131,72 @@ describe('ContentTypeFieldsPropertiesFormComponent', () => {
...
+            comp.form.get('searchable').setValue(true);
+
+            expect(comp.form.get('indexed').value).toBe(true);
+            expect(comp.form.get('indexed').disabled).toBe(true);
         });
 
-        xit('should save checked and auto checked checkbox', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed true when you select show in list', () => {
+            comp.form.get('indexed').setValue(false);
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts]: We need a test when indexed is undefined

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts [hunk 6]
```diff
@@ -88,70 +131,72 @@ describe('ContentTypeFieldsPropertiesFormComponent', () => {
...
+
+            expect(comp.form.get('indexed').value).toBe(true);
+            expect(comp.form.get('indexed').disabled).toBe(true);
         });
 
-        xit('should not unchecked indexed checkbox if unique checkbox is checked', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+        it('should set system indexed and required true when you select unique', () => {
+            comp.form.get('indexed').setValue(false);
+            comp.form.get('required').setValue(false);
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts]: We need some more test here:  when indexed is undefined but required not, when required is undefined but indexed not, both are undefined

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts [hunk 7]
```diff
@@ -88,70 +131,87 @@ describe('ContentTypeFieldsPropertiesFormComponent', () => {
...
+            startHostComponent();
+        });
+
+        it('should set unique and no break when indexed and required doesn\'t exist', () => {
+            comp.form.get('unique').setValue(true);
 
-        xit('should not unchecked indexed checkbox if unique checkbox is checked', () => {
-            // TODO: It needs a real mock of FieldPropertyService
+            expect(comp.form.get('indexed')).toBe(null);
+            expect(comp.form.get('required')).toBe(null);
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.spec.ts]: why just one test? I think we have other cases

**Pull request merged**